emeval/input/phone_view.py

```diff
@@ -343,27 +332,12 @@ def fill_motion_activity_df(self, storage_key):
             for phone_label in phone_map:
                 curr_calibration_ranges = phone_map[phone_label]["{}_ranges".format(storage_key)]
                 for r in curr_calibration_ranges:
-                    all_done = False
-                    motion_activity_entries = []
-                    curr_start_ts = r["start_ts"]
-                    prev_retrieved_count = 0
-
-                    while not all_done:
-                        print("About to retrieve data for %s from %s -> %s" % (phone_label, curr_start_ts, r["end_ts"]))
-                        curr_motion_activity_entries = self.spec_details.retrieve_data_from_server(phone_label, ["background/motion_activity"], curr_start_ts, r["end_ts"])
-                        print("Retrieved %d entries with timestamps %s..." % (len(curr_motion_activity_entries), [cle["metadata"]["write_ts"] for cle in curr_motion_activity_entries[0:10]]))
-                        if len(curr_motion_activity_entries) == 0 or len(curr_motion_activity_entries) == 1 or len(curr_motion_activity_entries) == prev_retrieved_count:
-                            all_done = True
-                        else:
-                            motion_activity_entries.extend(curr_motion_activity_entries)
-                            curr_start_ts = curr_motion_activity_entries[-1]["metadata"]["write_ts"]
-                            prev_retrieved_count = len(curr_motion_activity_entries)
-                    r["motion_activity_entries"] = motion_activity_entries
-                    motion_activity_df = pd.DataFrame([e["data"] for e in motion_activity_entries])
+                    r[BG_MOTION_ACTIVITY] = self.spec_details.retrieve_data(phone_label, [BG_MOTION_ACTIVITY], r["start_ts"], r["end_ts"])
+                    motion_activity_df = pd.DataFrame([e["data"] for e in r[BG_MOTION_ACTIVITY]])
```

Comment:
Oops. I think I first implemented this, and when I needed it again, pulled it out into a separate function to DRY. But then I forgot to remove this. Good catch! 😄